subsys/bluetooth/host/scan.c

```diff
@@ -460,8 +460,32 @@ static void le_adv_recv(bt_addr_le_t *addr, struct bt_le_scan_recv_info *info,
 				bt_lookup_id_addr(BT_ID_DEFAULT, addr));
 	}
 
+/*
+ * FIXME: The following assigns the address of the local `id_addr` object to
+ *        a field in the `info` object, which is provided by the caller and
+ *        outlives the local `id_addr` object, effectively creating a dangling
+ *        pointer.
+ *
+ *        While this is currently not a real problem because `info->addr` is
+ *        never dereferenced after the function returns, it does create a
+ *        dangling pointer nevertheless and needs to be fixed.
+ *
+ *        A proper fix will be provided as part of the GitHub issue
+ *        zephyrproject-rtos/zephyr#48459; until then, keep the dangling
+ *        pointer warning disabled to appease GCC 12.
+ */
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpragmas"
+#pragma GCC diagnostic ignored "-Wdangling-pointer"
+#endif
+
 	info->addr = &id_addr;
```

Comment:
Update: I read the referenced issue, and still think that the setting to NULL is a better initial fix (i.e. "option a") and then work on the allocation/lifetime as a follow up.

Reply:
> I may have missed the discussion leading to this particular fix

Please see https://github.com/zephyrproject-rtos/zephyr/issues/48459#issuecomment-1205184233.

This is not intended to be a "fix," but only a temporary workaround to get it building on the upcoming Zephyr SDK 0.15.0 release with GCC 12.

I did not implement the "option a" because I wanted to keep this PR as less intrusive as possible so as to not introduce any regressions (i.e. are we completely sure that `info->addr` is not referenced by the caller and setting it to `NULL` would not result in any regressions)?

If you can provide a proper fix (either implementing "option a" or "option b") before the Zephyr SDK 0.15.0 is mainlined (likely in one or two weeks), then this PR can be closed.

Reply:
> I did not implement the "option a" because I wanted to keep this PR as less intrusive as possible so as to not introduce any regressions (i.e. are we completely sure that info->addr is not referenced by the caller and setting it to NULL would not result in any regressions)?

Perceived intrusiveness is a subjecting thing. The preprocessor stuff you add in this PR appears more intrusive to me than a one line `var = NULL;` addition. I don't quite get what you mean by your regression comment: if the calling function tried to access the the pointer it would be accessing invalid memory since the stack frame of the function that set it is no longer valid. Setting the pointer to NULL is therefore a better initial fix IMO.

Reply:
@jhedberg Can you or one of the Bluetooth collaborators send a patch for that in the meantime then? I would prefer not to introduce any behavioural changes to this subsystem myself since I am not familiar enough with it.

Reply:
@stephanosio #48727

Reply:
@jhedberg Thanks!